src/rust/engine/src/cffi_build.rs

```diff
@@ -54,6 +56,16 @@ fn main() {
     println!("cargo:rustc-link-lib=static=stdc++");
   }
 
+  // Generate the scheduler.h bindings from the rust code in this crate.
+  let bindings_config_path = Path::new("cbindgen.toml");
+  mark_for_change_detection(&bindings_config_path);
+
+  let scheduler_file_path = Path::new("src/cffi/scheduler.h");
+  let crate_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
+  cbindgen::generate(crate_dir)
+    .expect("Unable to generate bindings")
+    .write_to_file(scheduler_file_path);
```

Comment:
Does this method return a `Result`? Not sure whether build scripts render warnings.

If it doesn't return a `Result`, then it looks like we're unwrap()ing everything in here, and there is no need for this `main` to return `Result`?

Reply:
I think the issue was that when I tried to use the `?` operator, it didn't have the right failure type for the return value of `main() -> std::io::Result<()>`. I definitely would like to remove `.unwrap()` calls like the env var one here given that we can wrap it all in a Result and will look to see if that works (and if a top-level Result renders failures).

Reply:
Understood. But you do not currently use the `?` operator.

Reply:
I have added some `From`s and removed all the `.unwrap()`s except for path strings, let me know if I should convert those too.